A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/A320_Neo_EICAS.js

```diff
@@ -156,16 +159,21 @@ class A320_Neo_EICAS extends Airliners.BaseEICAS {
 
     onUpdate() {
         let _deltaTime = this.getDeltaTime();
-
         super.onUpdate(_deltaTime);
 
         const selfTestCurrentKnobValue = SimVar.GetSimVarValue(this.isTopScreen ? "LIGHT POTENTIOMETER:92" : "LIGHT POTENTIOMETER:93", "number");
+        // ECAM all button
+        const ecamAllButtonState = SimVar.GetSimVarValue("L:A32NX_ECAM_ALL_Push_IsDown", "Bool");
+        if (ecamAllButtonState && this.ecamAllButtonTimerStarted) {
+            this.ecamAllButtonTimer -= _deltaTime;
+        }
         const knobChanged = (selfTestCurrentKnobValue >= 0.1 && this.selfTestLastKnobValue < 0.1);
 
         _deltaTime = this.updateThrottler.canUpdate(_deltaTime);
-        if (_deltaTime === -1 && !knobChanged) {
+        if (_deltaTime === -1 && !knobChanged && !ecamAllButtonState) {
```

Comment:
**issue**: having any condition other than `_deltaTime === -1` in this expression is problematic, as code will be executed below with a _deltaTime of `-1` which is never intended.

Reply:
Good point, I moved it further up to eliminate a possible -1 as delta time